### DISCUSSION TITLE
Allocate Stock Items for sales Orders?

### DISCUSSION THREAD

**Discussion by SparkeyinVA** [inventree/InvenTree#4876] [Q&A] [answered]
I am totally lost.  I am using V0.11.0.  I can create a Sales order, I can add shipment. I was able to create and make the required build orders.  As I complete outputs from the build order (parts are trackable with a serial number), I would like to add to a daily shipment.  I can not find a way to Allocate these parts to a shipment from the main program.  I was able to go to the Inventree Admin site administration screen and using the [Sales order allocations] I could allocate the parts to ship.  But I can not figure out that step in the main program that the regular user can do this.

**Comment by SchrodingersGat** [comment]
@SparkeyinVA a lot of questions there! Have you tried playing around on the [demo server](https://demo.inventree.org/) - it has a bunch of sales orders already set up so you might be able to get a better idea of how it all fits together

**Reply by SparkeyinVA** [reply]
@SchrodingersGat I will look at the demo server.  And will come back comeback if i still have questions.  I saw a post about examples and a walkthrough. Has that project ever been picked up?

**Reply by matmair** [reply]
Would be easier to answer if you had a link to the post

**Reply by SparkeyinVA** [reply]
> Would be easier to answer if you had a link to the post

@matmair  here is the issue I was referring to. https://github.com/inventree/inventree-docs/issues/211 
I believe a How to section would be helpful.

**Comment by SparkeyinVA** [comment]
@SchrodingersGat and @matmair I still need help to figure this out.  I am trying to understand how to Allocate Stock Items to a Pending shipment in a sales order.

I found this in the documentation.  https://docs.inventree.org/en/stable/sell/so/#allocate-stock-items  and it says "allocate stock items for that part to the sales order (click on  button)"  However there is no button to allocate stock.

If it matters the stock to ship is the completed output from a Build order tied to the sales order I am trying to ship from.

**Reply by SchrodingersGat** [reply, marked as answer]
Under the "line items" tab, where you have added individual parts to the sales order, click on the "allocate stock" button:

![image](https://github.com/inventree/InvenTree/assets/10080325/ac46a630-b0b9-4c98-a67b-78afe0c50bf4)

This gives you a dialog to select stock and shipment:

![image](https://github.com/inventree/InvenTree/assets/10080325/538ee745-eaf1-4aa2-a76c-8c8a0fcfebde)

**Reply by SparkeyinVA** [reply]
@SchrodingersGat 
For some reason on my copy that button is not there.
![Screenshot 2023-05-23 084010](https://github.com/inventree/InvenTree/assets/50155189/41d8190c-4d3f-4fe6-b6f2-53a42812673f)

As you can see there are three that are 3 that are available.

**Reply by SchrodingersGat** [reply]
@SparkeyinVA looks like a bug in 0.11.0 (which has been fixed in master).

Ref: https://github.com/inventree/InvenTree/pull/4753

I will back-port this for 0.11.2 version

**Reply by SparkeyinVA** [reply]
@SchrodingersGat Okay, thank you.  I am using the stable for the installation, is there a way to have it update to 0.11.1 or 0.11.2 or would I need to install using master?

**Reply by SchrodingersGat** [reply]
`0.11.1` is out, so you can update to that. `0.11.2` is not out yet though

**Reply by SparkeyinVA** [reply]
> Under the "line items" tab, where you have added individual parts to the sales order, click on the "allocate stock" button:
> 
> ![image](https://user-images.githubusercontent.com/10080325/240273207-ac46a630-b0b9-4c98-a67b-78afe0c50bf4.png)
> 
> This gives you a dialog to select stock and shipment:
> 
> ![image](https://user-images.githubusercontent.com/10080325/240273263-538ee745-eaf1-4aa2-a76c-8c8a0fcfebde.png)

I would add one additional point related to this issue.  After updating when logging in on the server, the issue was corrected. The issue was still observed after the update to V0.11.2 on a remote computer using the same login.  The issue was corrected after clearing the cache on the remote computer's browser. 

Thanks again to @SchrodingersGat and @matmair for your assistance with this issue.
